src/forms/BookingDatesForm/BookingDatesForm.js

```diff
@@ -84,54 +104,67 @@ export class BookingDatesFormComponent extends Component {
 
           const bookingStartLabel = intl.formatMessage({
             id: 'BookingDatesForm.bookingStartTitle',
           });
-          const bookingEndLabel = intl.formatMessage({ id: 'BookingDatesForm.bookingEndTitle' });
-          const requiredMessage = intl.formatMessage({ id: 'BookingDatesForm.requiredDate' });
+          const bookingEndLabel = intl.formatMessage({
+            id: 'BookingDatesForm.bookingEndTitle',
+          });
+          const requiredMessage = intl.formatMessage({
+            id: 'BookingDatesForm.requiredDate',
+          });
           const startDateErrorMessage = intl.formatMessage({
             id: 'FieldDateRangeInput.invalidStartDate',
           });
           const endDateErrorMessage = intl.formatMessage({
             id: 'FieldDateRangeInput.invalidEndDate',
           });
           const timeSlotsError = fetchTimeSlotsError ? (
-            <p className={css.timeSlotsError}>
+            <p className={css.sideBarError}>
               <FormattedMessage id="BookingDatesForm.timeSlotsError" />
             </p>
           ) : null;
 
-          // This is the place to collect breakdown estimation data. See the
-          // EstimatedBreakdownMaybe component to change the calculations
-          // for customized payment processes.
+          // This is the place to collect breakdown estimation data.
+          // Note: lineItems are calculated and fetched from FTW backend
+          // so we need to pass only booking data that is needed otherwise
+          // If you have added new fields to the form that will affect to pricing,
+          // you need to add the values to handleOnChange function
           const bookingData =
             startDate && endDate
               ? {
                   unitType,
-                  unitPrice,
                   startDate,
                   endDate,
-
-                  // NOTE: If unitType is `line-item/units`, a new picker
-                  // for the quantity should be added to the form.
-                  quantity: 1,
                 }
               : null;
-          const bookingInfo = bookingData ? (
-            <div className={css.priceBreakdownContainer}>
-              <h3 className={css.priceBreakdownTitle}>
-                <FormattedMessage id="BookingDatesForm.priceBreakdownTitle" />
-              </h3>
-              <EstimatedBreakdownMaybe bookingData={bookingData} />
-            </div>
+
+          const bookingInfo =
+            bookingData && lineItems && !fetchLineItemsError ? (
+              <div className={css.priceBreakdownContainer}>
+                <h3 className={css.priceBreakdownTitle}>
+                  <FormattedMessage id="BookingDatesForm.priceBreakdownTitle" />
+                </h3>
+                <EstimatedBreakdownMaybe bookingData={bookingData} lineItems={lineItems} />
+              </div>
+            ) : fetchLineItemsInProgress ? (
+              <IconSpinner className={css.spinner} />
```

Comment:
Hmm, this nested ternary is a bit hard to understand. I know it's already quite common in the FTW codebase, but I'm not sure if `bookingData && lineItems && !fetchLineItemsError` and `fetchLineItemsInProgress` can both be true. Then the estimation would be rendered since it's checked first, even if the fetch is in progress.

This is related to the in-progress fetch order mentioned in the previous comment.